src/main/java/org/whispersystems/textsecuregcm/providers/RedisClientFactory.java

```diff
@@ -33,23 +33,25 @@
 import redis.clients.jedis.JedisPool;
 import redis.clients.jedis.JedisPoolConfig;
 import redis.clients.jedis.Protocol;
+import redis.clients.util.JedisURIHelper;
 
 public class RedisClientFactory implements RedisPubSubConnectionFactory {
 
   private final Logger logger = LoggerFactory.getLogger(RedisClientFactory.class);
 
   private final String    host;
   private final int       port;
+  private final String    password;
   private final ReplicatedJedisPool jedisPool;
 
   public RedisClientFactory(String url, List<String> replicaUrls) throws URISyntaxException {
     JedisPoolConfig poolConfig = new JedisPoolConfig();
     poolConfig.setTestOnBorrow(true);
 
     URI redisURI = new URI(url);
-
     this.host      = redisURI.getHost();
     this.port      = redisURI.getPort();
+    this.password  = JedisURIHelper.getPassword(redisURI);
 
     JedisPool       masterPool   = new JedisPool(poolConfig, host, port, Protocol.DEFAULT_TIMEOUT, null);
```

Comment:
When creating pools shouldn't you also include the password here?

Reply:
Sorry I messed up this branch :) I created a new one to not loose much time, and added password to JedisPool constructors too.